### PR TITLE
PSROIPool + Dispatcher + Autocast + Code Cleanup

### PR DESCRIPTION
**Pull request by datumbox**
part of #2797 and #2796

Implements Dispatcher + Autocast for PSROIPool.

I'm also doing some clean up on the original code such as:
- Remove primitive const declaration from method names.
- Using references when possible.
- Fix variable naming.

### CODE REVIEW

File: torchvision/csrc/PSROIPool.h [hunk 1]
```diff
@@ -117,13 +127,77 @@ class PSROIPoolFunction : public torch::autograd::Function<PSROIPoolFunction> {
...
+      rois,
+      channel_mapping,
+      spatial_scale,
+      pooled_height,
+      pooled_width,
+      batch_size,
+      channels,
+      height,
+      width)[0];
+}
```
Comment (fmassa) [inline torchvision/csrc/PSROIPool.h:203]: nit: newline

**Review by fmassa — APPROVED** [review]
Looks great, thanks a lot!

File: torchvision/csrc/PSROIPool.h (hunk 1 above)
Comment (datumbox) [reply torchvision/csrc/PSROIPool.h]: done

**Review by fmassa — APPROVED** [review]
Merging, thanks a lot!

For the future, let's use ghstack to avoid adding unrelated changes to the PR in a way that helps with separate feature development

**Pull request merged**